e2e_tests.py

```diff
@@ -4,19 +4,34 @@
 from utils import type_check, timems
 import urllib.parse
 from config import config
+import sys
+import threading
 
-host = 'http://localhost:' + str (config ['port']) + '/'
+# USAGE: python e2e_tests.py [CONCURRENT_TESTS] [alpha|test]
```

Comment:
If all is well, `argparse` should actually be telling you this if you forget.

So the other advantage of doing that is that your tool becomes self-documenting, and also responds to `e2e_tests.py --help`.

Reply:
See and it does:

```
$  python3 e2e_tests.py --help
usage: e2e_tests.py [-h] [--concurrent CONCURRENT] [--host HOST]

optional arguments:
  -h, --help            show this help message and exit
  --concurrent CONCURRENT
                        how many tests to run at the same time (stress
                        testing), default is 1
  --host HOST           Host against which to run the tests (optional),
                        default is localhost

```

Reply:
True! But I wanted to leave a comment in the code with the usage, particularly for those not acquainted with `argparse`.

I have to say, adding `argparse` was a great idea!